code/modules/mob/living/carbon/xenomorph/damage_procs.dm

```diff
@@ -112,6 +112,11 @@
 	updatehealth()
 	return TRUE
 
+/mob/living/carbon/Xenomorph/Hunter/apply_damage(damage = 0, damagetype = BRUTE, def_zone = null, blocked = 0, used_weapon = null, sharp = 0, edge = 0)
+	. = ..()
+	if(damage > 15 && stealth) //Any significant damage causes us to break stealth
```

Comment:
You can substitute the check for a caste-specific var for a proc.
Like `is_stealthed()`
You do that with:
```
/mob/living/carbon/Xenomorph/proc/is_stealthed()
	return FALSE
/mob/living/carbon/Xenomorph/Hunter/is_stealthed()
	if(stealth)
		return TRUE
	else
		return FALSE
```